### PR TITLE
Fix EZP-23971: if a user is removed while logged in, the session for tha...

### PR DESCRIPTION
**Pull request by glye**
...t user will cause a fatal error

Test if the user exists by reloading it. If it doesn't, login anonymous user.

This is certainly NOT the way to fix this issue (though it does fix the reported bug), but I could use some feedback on a better way to fix it.

### CODE REVIEW

**Comment by yannickroger** [comment]
as requested : NOT +1 :)

**Comment by glye** [comment]
They say "All PR is good PR" but this PR is proving them wrong :)

File: eZ/Publish/Core/MVC/Symfony/Security/EventListener/SecurityListener.php [hunk 1]
```diff
@@ -213,6 +214,15 @@ public function onKernelRequest( GetResponseEvent $event )
             return;
         }
 
+        try
+        {
+            $this->repository->getUserService()->loadUser( $this->repository->getCurrentUser()->contentInfo->id );
+        }
+        catch ( NotFoundException $e )
+        {
+            $this->repository->setCurrentUser( $this->repository->getUserService()->loadUser( 10 ) );
```
Comment (crevillo) [inline eZ/Publish/Core/MVC/Symfony/Security/EventListener/SecurityListener.php]: nitpick... are we sure that anonymous user are always 10? i know some customers who accidentally removed the anonymous user :)

Comment (glye) [inline eZ/Publish/Core/MVC/Symfony/Security/EventListener/SecurityListener.php]: Yes, this is one of the bits that are NOT ok. Will certainly improve this, if by any chance I get the green light for the general idea of the fix.

Comment (crevillo) [inline eZ/Publish/Core/MVC/Symfony/Security/EventListener/SecurityListener.php]: ok. my doubt is, if the user has been deleted what ``` $this->repository->getCurrentUser()``` returns? something from the session?

**Comment by andrerom** [comment]
I guess this approach might have quite some performance impact, lets ping @lolautruche for alternative approaches maybe.

**Comment by glye** [comment]
@andrerom Yes, my main concern is performance, second is API abuse (possibly) and finally the hardcoded anon ID 10 (easy fix).

File: eZ/Publish/Core/MVC/Symfony/Security/EventListener/SecurityListener.php (hunk 1 above)
Comment (glye) [inline eZ/Publish/Core/MVC/Symfony/Security/EventListener/SecurityListener.php]: @crevillo Yes, it returns the new stack user object as usual, no problem there. This is the source of the bug actually: New stack doesn't know that the user has been deleted, and will happily continue to use the new stack user objects (until it discovers that the user has no role permissions).

**Pull request closed without merging**